src/core/handlers/RequestHandler.ts

```diff
@@ -184,6 +186,36 @@ export abstract class RequestHandler<
     })
   }
 
+  private static cookiesForRequestCache = new WeakMap<
+    Request,
+    Record<string, string>
+  >()
+  protected static parseAllRequestCookiesOrGetFromCache(
+    request: Request,
+  ): Record<string, string> {
+    let cookies = RequestHandler.cookiesForRequestCache.get(request)
+    if (!cookies) {
+      cookies = getAllRequestCookies(request)
+      RequestHandler.cookiesForRequestCache.set(request, cookies)
+    }
+    return cookies
+  }
+
+  private static matchRequestUrlCache = new Map<string, Match>()
+  protected static matchRequestURLOrGetMatchFromCache(
```

Comment:
URL matching can be computed _once_ per string comparison of URL/match/baseUrl - and reused for subsequent requests across all handlers.  There's some memory cost to this map, but even at very large sizes it should be fine